syntax = "proto2";

import "nanopb.proto";
message AudioFile {
    optional bytes file_id = 1 [(nanopb).type = FT_POINTER];
    optional AudioFormat format = 2;
}

message Restriction {
    optional string countries_allowed = 0x2 [(nanopb).type = FT_POINTER];
    optional string countries_forbidden = 0x3 [(nanopb).type = FT_POINTER];
}

message Image {
    optional bytes file_id = 0x1 [(nanopb).type = FT_POINTER];
}

message ImageGroup {
    repeated Image image = 0x1 [(nanopb).type = FT_POINTER];
}

message Album {
    optional bytes gid = 0x1 [(nanopb).type = FT_POINTER];
    optional string name = 0x2 [(nanopb).type = FT_POINTER];
    optional ImageGroup cover_group = 0x11;
}

message Artist {
    optional bytes gid = 0x1 [(nanopb).type = FT_POINTER];
    optional string name = 0x2 [(nanopb).type = FT_POINTER];
}

message Track {
    optional bytes gid = 1 [(nanopb).type = FT_POINTER];
    optional string name = 2 [(nanopb).type = FT_POINTER];
    optional sint32 number = 5;
    optional sint32 disc_number = 6;
    optional sint32 duration = 0x7;
    optional Album album = 0x3;
    repeated Artist artist = 0x4 [(nanopb).type = FT_POINTER];
    repeated Restriction restriction = 0xb [(nanopb).type = FT_POINTER];
    repeated AudioFile file = 0xc [(nanopb).type = FT_POINTER];
    repeated Track alternative = 0xd [(nanopb).type = FT_POINTER];
}

message Episode {
    optional bytes gid = 1 [(nanopb).type = FT_POINTER];
    optional string name = 2 [(nanopb).type = FT_POINTER];
    optional sint32 duration = 7;
    optional sint32 number = 65;
    repeated AudioFile file = 12 [(nanopb).type = FT_POINTER];
    repeated Restriction restriction = 0x4B [(nanopb).type = FT_POINTER];
    optional ImageGroup covers = 0x44;
}

enum AudioFormat {
    OGG_VORBIS_96 = 0;
    OGG_VORBIS_160 = 1;
    OGG_VORBIS_320 = 2;
    MP3_256 = 3;
    MP3_320 = 4;
    MP3_160 = 5;
    MP3_96 = 6;
    MP3_160_ENC = 7;
    AAC_24 = 8;
    AAC_48 = 9;
}
